tools/depmod.c

```diff
@@ -322,52 +324,104 @@ static int index__haschildren(const struct index_node *node)
 	return node->first < INDEX_CHILDMAX;
 }
 
-/* Recursive post-order traversal
+static uint32_t index_get_mask(const struct index_node *node)
+{
+	uint32_t mask = 0;
 
-   Pre-order would make for better read-side buffering / readahead / caching.
-   (post-order means you go backwards in the file as you descend the tree).
-   However, index reading is already fast enough.
-   Pre-order is simpler for writing, and depmod is already slow.
- */
-static uint32_t index_write__node(const struct index_node *node, FILE *out)
+	if (!node)
+		return mask;
```

Comment:
Removed, didn't add `__nonnull_all_` because I would have to add headers. It's all static, so hopefully "we can see it."

Reply:
Hm, for whatever reason it didn't work at first. Probably mistyped something. Adding this would make sense if it's done everywhere, since pretty much all these index functions do not expect NULL (anymore). Sounds like a general cleanup which could be done -- in another PR, I think.